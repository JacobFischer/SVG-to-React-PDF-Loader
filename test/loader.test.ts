import compile from "./webpack-compile";

describe("Webpack loader", () => {
    test("Inserts name and outputs JavaScript", async () => {
        const stats = await compile("svgs/paint.svg");

        const output = stats.toJson();
        const { modules } = output;

        expect(modules).toBeDefined();
        expect(modules && modules[0].source).toBe(
            `export default "Hey Alice!"`,
        );
    });
});
