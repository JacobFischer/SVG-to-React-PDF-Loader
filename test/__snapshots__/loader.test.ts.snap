// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack loader Accepts a module option 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.default = void 0;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _react = _interopRequireDefault(require(\\"../node_modules/react/index.js\\"));

var mod = _interopRequireDefault(require(\\"@react-pdf/renderer\\"));

function component(str) {
    if (mod[str]) {
        return mod[str];
    }

    if (mod.default && mod.default[str]) {
        return mod.default[str];
    }

    return function NothingForTag() {
        return null;
    }
}

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var XmlAsReactComponent = function XmlAsReactComponent(props) {
    return (
        /*#__PURE__*/_react.default.createElement(component(\\"Svg\\"), _extends({ \\"height\\": \\"100\\", \\"width\\": \\"100\\" }, props),
            /*#__PURE__*/_react.default.createElement(component(\\"Circle\\"), { \\"cx\\": \\"50\\", \\"cy\\": \\"50\\", \\"fill\\": \\"red\\", \\"r\\": \\"40\\", \\"stroke\\": \\"black\\", \\"stroke-width\\": \\"3\\" }),
        )
    );
}

var rootAttributes = {\\"height\\":\\"100\\",\\"width\\":\\"100\\"};
exports.rootAttributes = rootAttributes;

var _default = XmlAsReactComponent;
exports.default = _default;
"
`;

exports[`Webpack loader Transforms inner text within SVGs 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.default = void 0;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _react = _interopRequireDefault(require(\\"../../node_modules/react/index.js\\"));

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var XmlAsReactComponent = function XmlAsReactComponent(props) {
    return (
        /*#__PURE__*/_react.default.createElement(\\"svg\\", _extends({ \\"viewBox\\": \\"0 0 240 80\\", \\"xmlns\\": \\"http://www.w3.org/2000/svg\\" }, props),
            /*#__PURE__*/_react.default.createElement(\\"style\\", null,
                \\"\\\\n    .small { font: italic 13px sans-serif; }\\\\n    .heavy { font: bold 30px sans-serif; }\\\\n\\\\n    /* Note that the color of the text is set with the    *\\\\n     * fill property, the color property is for HTML only */\\\\n    .Rrrrr { font: italic 40px serif; fill: red; }\\\\n  \\",
            ),
            /*#__PURE__*/_react.default.createElement(\\"text\\", { \\"class\\": \\"small\\", \\"x\\": \\"20\\", \\"y\\": \\"35\\" },
                \\"My\\",
            ),
            /*#__PURE__*/_react.default.createElement(\\"text\\", { \\"class\\": \\"heavy\\", \\"x\\": \\"40\\", \\"y\\": \\"35\\" },
                \\"cat\\",
            ),
            /*#__PURE__*/_react.default.createElement(\\"text\\", { \\"class\\": \\"small\\", \\"x\\": \\"55\\", \\"y\\": \\"55\\" },
                \\"is\\",
            ),
            /*#__PURE__*/_react.default.createElement(\\"text\\", { \\"class\\": \\"Rrrrr\\", \\"x\\": \\"65\\", \\"y\\": \\"55\\" },
                \\"Grumpy!\\",
            ),
        )
    );
}

var rootAttributes = {\\"viewBox\\":\\"0 0 240 80\\",\\"xmlns\\":\\"http://www.w3.org/2000/svg\\"};
exports.rootAttributes = rootAttributes;

var _default = XmlAsReactComponent;
exports.default = _default;
"
`;

exports[`Webpack loader Transforms minimal SVGs 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.default = void 0;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _react = _interopRequireDefault(require(\\"../../node_modules/react/index.js\\"));



var XmlAsReactComponent = function XmlAsReactComponent(props) {
    return (
        /*#__PURE__*/_react.default.createElement(\\"svg\\", props)
    );
}

var rootAttributes = {};
exports.rootAttributes = rootAttributes;

var _default = XmlAsReactComponent;
exports.default = _default;
"
`;

exports[`Webpack loader Works as a webpack loader 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.default = void 0;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _react = _interopRequireDefault(require(\\"../node_modules/react/index.js\\"));

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var XmlAsReactComponent = function XmlAsReactComponent(props) {
    return (
        /*#__PURE__*/_react.default.createElement(\\"svg\\", _extends({ \\"height\\": \\"100\\", \\"width\\": \\"100\\" }, props),
            /*#__PURE__*/_react.default.createElement(\\"circle\\", { \\"cx\\": \\"50\\", \\"cy\\": \\"50\\", \\"fill\\": \\"red\\", \\"r\\": \\"40\\", \\"stroke\\": \\"black\\", \\"stroke-width\\": \\"3\\" }),
        )
    );
}

var rootAttributes = {\\"height\\":\\"100\\",\\"width\\":\\"100\\"};
exports.rootAttributes = rootAttributes;

var _default = XmlAsReactComponent;
exports.default = _default;
"
`;
